pysat/tests/test_files.py

```diff
@@ -272,171 +210,366 @@ def test_inequality_modified_object(self):
+
+        """
+
+        # Truth dates
+        dates = pysat.utils.time.create_date_range(self.start, self.stop, freq)
+        if len(dates) > 100000:
+            self.stop = self.start + dt.timedelta(days=1)
+            dates = pysat.utils.time.create_date_range(self.start, self.stop,
+                                                       freq)
+
+        # Create a bunch of files by year and doy
+        create_files(self.testInst, self.start, self.stop, freq=freq,
+                     root_fname=root_fname, version=self.version,
+                     use_doy=use_doy)
+
+        # Use `from_os` function to get pandas Series of files and dates
         files = pysat.Files.from_os(data_path=self.testInst.files.data_path,
-                                    format_str=root_fname)
-        # check overall length
-        assert len(files) == (365 + 366)
-        # check specific dates
-        assert pds.to_datetime(files.index[0]) == dt.datetime(2008, 1, 1)
-        assert pds.to_datetime(files.index[365]) == dt.datetime(2008, 12, 31)
-        assert pds.to_datetime(files.index[-1]) == dt.datetime(2009, 12, 31)
-
-    def test_year_doy_files_no_gap_in_name_directly_call_from_os(self):
-        """Files.from_os generates file list for date w/o delimiter"""
-        # create a bunch of files by year and doy
-        root_fname = ''.join(('pysat_testing_junk_{year:04d}{day:03d}_stuff.',
-                              'pysat_testing_file'))
+                                    format_str=root_fname, delimiter=delimiter)
+
+        # Ensure sorted and increasing
+        assert files.index.is_monotonic_increasing
+
+        # Check overall length
+        assert len(files) == len(dates)
+
+        # Check specific date
+        assert np.all(files.index == dates)
+
+        return
+
+    @pytest.mark.parametrize("root_fname,root_pname",
+                             [[''.join(['pysat_1234567_junk_{year:04d}_gold_',
+                                       '{day:03d}_stuff.pysat_testing_file']),
+                               ''.join(['pysat_{code:7s}_junk_{year:04d}_gold_',
+                                        '{day:03d}_stuff.pysat_testing_file'])],
+                              [''.join(['pysat_1234567_junk_{year:04d}_gold_',
+                                        '{day:03d}_stuff.pysat_testing_file']),
+                               ''.join(['pysat_123{code:4s}_junk_{year:04d}_',
+                                        'gold_{day:03d}_stuff.pysat_testing',
+                                        '_file'])],
+                              [''.join(['pysat_1234567_junk_{year:04d}_gold_',
+                                        '{day:03d}_stuff.pysat_testing_file']),
+                               ''.join(['{code:5s}_{code2:7s}_junk_{year:04d}',
+                                        '_gold_{day:03d}_stuff.pysat_testing',
+                                        '_file'])]
+                              ])
```

Comment:
```suggestion
    @pytest.mark.parametrize(
        "root_fname,root_pname",
        [['pysat_1234567_junk_{year:04d}_gold_{day:03d}_stuff',
          'pysat_{code:7s}_junk_{year:04d}_gold_{day:03d}_stuff'],
         ['pysat_1234567_junk_{year:04d}_gold_{day:03d}_stuff',
          'pysat_123{code:4s}_junk_{year:04d}_gold_{day:03d}_stuff'],
         ['pysat_1234567_junk_{year:04d}_gold_{day:03d}_stuff',
          '{code:5s}_{code2:7s}_junk_{year:04d}_gold_{day:03d}_stuff']])
```
This plus suggestion below works locally, and would be cleaner to read